packages/client/components/DiscussionThread.tsx

```diff
@@ -69,12 +69,31 @@ export default createFragmentContainer(DiscussionThread, {
   viewer: graphql`
     fragment DiscussionThread_viewer on User {
       meeting(meetingId: $meetingId) {
-        ... on RetrospectiveMeeting {
+        ... on NewMeeting {
           ...DiscussionThreadInput_meeting
           ...DiscussionThreadList_meeting
           endedAt
           replyingToCommentId
-          reflectionGroup(reflectionGroupId: $reflectionGroupId) {
+        }
+        ... on RetrospectiveMeeting {
+          threadSource: reflectionGroup(reflectionGroupId: $threadSourceId) {
+            id
```

Comment:
the fact that both `agendaItem` and `reflectionGroup` have a `thread` connection tells me that we could probably DRY this up a bit. 

To do so, try creating a `ThreadSource` as a  `GraphQLInterfaceType` that implements the `thread` connection and then make agendaItem & reflectionGroup implement the `ThreadSource` interface. See https://github.com/ParabolInc/parabol/blob/8c68bf771a350b31c7948e6bd11c5c110e30c977/packages/server/graphql/types/Reactable.ts for inspiration.

Reply:
i totally agree. thanks for suggesting a way to implement that! `ThreadSource` interface type has been created in the branch with `AgendaItem` and `ReflectionGroup` iimplementing the interface.